packages/tailwindcss/src/css-functions.ts

```diff
@@ -51,15 +64,70 @@ function spacing(designSystem: DesignSystem, value: string, ...rest: string[]) {
   return `calc(${multiplier} * ${value})`
 }
 
-function theme(designSystem: DesignSystem, path: string, ...fallback: string[]) {
+function theme(
+  designSystem: DesignSystem,
+  source: AstNode,
+  path: string,
+  ...fallback: string[]
+): string {
   if (!path.startsWith('--')) {
     throw new Error(`The --theme(…) function can only be used with CSS variables from your theme.`)
   }
 
-  return legacyTheme(designSystem, path, ...fallback)
+  let inline = false
+
+  // Handle `--theme(… inline)` to force inline resolution
+  if (path.endsWith(' inline')) {
+    inline = true
+    path = path.slice(0, -7)
+  }
+
+  // If the `--theme(…)` function is used within an at-rule (e.g. `@media (width >= --theme(…)))`,
+  // we have to always inline the result since CSS does not support CSS variables in these positions
+  if (source.kind === 'at-rule') {
+    inline = true
+  }
+
+  let resolvedValue = designSystem.resolveThemeValue(path, inline)
+
+  if (!resolvedValue) {
+    if (fallback.length > 0) return fallback.join(', ')
+    throw new Error(
+      `Could not resolve value for theme function: \`theme(${path})\`. Consider checking if the variable name is correct or provide a fallback value to silence this error.`,
+    )
+  }
+
+  if (fallback.length === 0) {
+    return resolvedValue
+  }
+
+  // Inject the fallback…
+  //
+  // - …as the value if the value returned is `initial`
+  // - …expect any `initial` fallbacks on `var(…)`, `theme(…)`, or `--theme(…)`
```

Comment:
Is this supposed to say `except`?

This entire comment is kinda hard to parse for me

Reply:
I reworked this part, thanks